Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -694,7 +694,7 @@ private Implementation[] GetPrioritizedImplementations(Program program)
       return GetPrioritizedImplementations(program).Select(implementation => new ImplementationTask(this, processedProgram, implementation)).ToList();
     }
 
-    public async Task<VerificationResult> VerifyImplementationWithLargeStackScheduler(
+    public async Task<Task<VerificationResult>> EnqueueVerifyImplementation(
```

Comment:
```suggestion
    /// The outer task is to wait for a semaphore to let verification start
    /// The inner task is the actual verification of the implementation
    public async Task<Task<VerificationResult>> EnqueueVerifyImplementation(
```
```suggestion
    public async Task<Task<VerificationResult>> EnqueueVerifyImplementation(
```

When something is weird, such as a nested task here, I'd strongly encourage adding a comment to the method, even if it is private. It took me substantial time to figure out why two nested tasks were needed.

Reply:
Good suggestion, thanks.